protocol/java/org/openyolo/protocol/AuthenticationMethod.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2016 The OpenYOLO Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.openyolo.protocol;
+
+import static org.valid4j.Assertive.require;
+
+import android.support.annotation.NonNull;
+import java.io.IOException;
+import org.openyolo.protocol.internal.CustomMatchers;
+
+
+/**
+ * A URI-based representation of an authentication method, which describes how a credential is
+ * used to authenticate a user.
+ */
+public final class AuthenticationMethod implements Comparable<AuthenticationMethod> {
+
+    /**
+     * Creates an authentication method from its protocol buffer equivalent, in byte form.
+     * @throws IOException if the protocol buffer cannot be parsed.
+     */
+    public static AuthenticationMethod fromProtobufBytes(byte[] protoBufBytes) throws IOException {
+        return fromProtobuf(Protobufs.AuthenticationMethod.parseFrom(protoBufBytes));
+    }
+
+    /**
+     * Creates an authentication method from its protocol buffer equivalent.
+     */
+    public static AuthenticationMethod fromProtobuf(Protobufs.AuthenticationMethod authMethod) {
+        return new AuthenticationMethod(authMethod.getUri());
+    }
+
+    private final String mUri;
+
+    /**
+     * Creates an authentication method from a URI string.
+     */
+    public AuthenticationMethod(@NonNull String authMethodUriStr) {
+        mUri = require(authMethodUriStr, CustomMatchers.isValidAuthenticationMethod());
+    }
+
+    public boolean isFederated() {
+        return mUri.startsWith("https://");
```

Comment:
1) Move string to a constant.
2) Missing Javadocs
3) I'm hesitant adding this as it is not defined in the protocol.  Since providers are able to save with any authentication method, it's easy to introduce cases where this is not true.